### PR TITLE
Add thread cycle with constuctor

### PR DESCRIPTION
**Pull request by fubali01**

### CODE REVIEW

File: train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java [hunk 1]
```diff
@@ -4,6 +4,21 @@
...
 public class TrainControllerImpl implements TrainController {
 
+	public TrainControllerImpl() {
+		Thread t = new Thread() {
+			@Override
+			public void run() {
+				followSpeed();
+				try {
+					Thread.sleep(1000);
+				} catch (InterruptedException e) {
```
Comment (julilly) [inline train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java:14]: nice exception handling

**Review by julilly — APPROVED** [review]

File: train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java [hunk 2]
```diff
@@ -4,6 +4,21 @@
 
 public class TrainControllerImpl implements TrainController {
 
+	public TrainControllerImpl() {
+		Thread t = new Thread() {
```
Comment (julilly) [inline train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java:8]: give the Thread a meaningful name

**Review by julilly — CHANGES_REQUESTED** [review]

File: train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java (hunk 1 above)
Comment (fubali01) [reply train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java:14]: not really a problem

File: train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java (hunk 2 above)
Comment (fubali01) [reply train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java:8]: ok

**Pull request merged**